File: jar/src/main/java/jline/lib/mom/SimpleExample.java
package jline.lib.mom;

import jline.lib.mom.util.MomUtils;
import jline.util.Maths;
import jline.util.matrix.Matrix;

public final class SimpleExample {
    private SimpleExample() {}

    public static void main(String[] args) {
        System.out.println("=== MOM Solver Integration Test ===\n");

        // Test 1: Using jline.util.Maths functions
        System.out.println("Test 1: Binomial coefficient (10 choose 3)");
        double binCoeff = Maths.binomialCoeff(10, 3);
        System.out.println("Result: " + binCoeff);

        // Test 2: Using multichoose
        System.out.println("\nTest 2: Multichoose(3, 2)");
        Matrix multiChoose = Maths.multichoose(3.0, 2.0);
        System.out.println("Results (" + multiChoose.getNumRows() + " combinations):");
        for (int i = 0; i < multiChoose.getNumRows(); i++) {
            System.out.print("  [");
            for (int j = 0; j < multiChoose.getNumCols(); j++) {
                System.out.print((int) multiChoose.get(i, j));
                if (j < multiChoose.getNumCols() - 1) System.out.print(", ");
            }
            System.out.println("]");
        }

        // Test 3: Using Matrix operations
        System.out.println("\nTest 3: Matrix operations");
        Matrix matrix1 = new Matrix(2, 2);
        matrix1.set(0, 0, 1.0);
        matrix1.set(0, 1, 2.0);
        matrix1.set(1, 0, 3.0);
        matrix1.set(1, 1, 4.0);

        Matrix matrix2 = new Matrix(2, 2);
        matrix2.set(0, 0, 5.0);
        matrix2.set(0, 1, 6.0);
        matrix2.set(1, 0, 7.0);
        matrix2.set(1, 1, 8.0);

        Matrix result = matrix1.mult(matrix2);
        System.out.println("Matrix multiplication result:");
        for (int i = 0; i < result.getNumRows(); i++) {
            System.out.print("  [");
            for (int j = 0; j < result.getNumCols(); j++) {
                System.out.print(result.get(i, j));
                if (j < result.getNumCols() - 1) System.out.print(", ");
            }
            System.out.println("]");
        }

        // Test 4: MOM-specific utilities
        System.out.println("\nTest 4: MOM utilities");
        int[][] combinations = new int[][] {
                {2, 0, 0},
                {1, 1, 0},
                {0, 2, 0},
                {1, 0, 1},
                {0, 1, 1},
                {0, 0, 2}
        };

        System.out.println("Original combinations:");
        for (int[] c : combinations) {
            System.out.println("  " + java.util.Arrays.toString(c));
        }

        int[][] sorted = MomUtils.sortByNnzPos(combinations);
        System.out.println("\nSorted by non-zeros:");
        for (int[] c : sorted) {
            System.out.println("  " + java.util.Arrays.toString(c));
        }

        System.out.println("\n=== All tests completed successfully ===");
    }
}

File: jar/src/main/kotlin/jline/lib/mom/SimpleExample.kt
package jline.lib.mom

import jline.util.Maths
import jline.util.matrix.Matrix

fun main() {
    println("=== MOM Solver Integration Test ===\n")
    
    // Test 1: Using jline.util.Maths functions
    println("Test 1: Binomial coefficient (10 choose 3)")
    val binCoeff = Maths.binomialCoeff(10, 3)
    println("Result: $binCoeff")
    
    // Test 2: Using multichoose
    println("\nTest 2: Multichoose(3, 2)")
    val multiChoose = Maths.multichoose(3.0, 2.0)
    println("Results (${multiChoose.getNumRows()} combinations):")
    for (i in 0 until multiChoose.getNumRows()) {
        print("  [")
        for (j in 0 until multiChoose.getNumCols()) {
            print("${multiChoose.get(i, j).toInt()}")
            if (j < multiChoose.getNumCols() - 1) print(", ")
        }
        println("]")
    }
    
    // Test 3: Using Matrix operations
    println("\nTest 3: Matrix operations")
    val matrix1 = Matrix(2, 2)
    matrix1.set(0, 0, 1.0)
    matrix1.set(0, 1, 2.0)
    matrix1.set(1, 0, 3.0)
    matrix1.set(1, 1, 4.0)
    
    val matrix2 = Matrix(2, 2)
    matrix2.set(0, 0, 5.0)
    matrix2.set(0, 1, 6.0)
    matrix2.set(1, 0, 7.0)
    matrix2.set(1, 1, 8.0)
    
    val result = matrix1.mult(matrix2)
    println("Matrix multiplication result:")
    for (i in 0 until result.getNumRows()) {
        print("  [")
        for (j in 0 until result.getNumCols()) {
            print("${result.get(i, j)}")
            if (j < result.getNumCols() - 1) print(", ")
        }
        println("]")
    }
    
    // Test 4: MOM-specific utilities
    println("\nTest 4: MOM utilities")
    val combinations = arrayOf(
        intArrayOf(2, 0, 0),
        intArrayOf(1, 1, 0),
        intArrayOf(0, 2, 0),
        intArrayOf(1, 0, 1),
        intArrayOf(0, 1, 1),
        intArrayOf(0, 0, 2)
    )
    
    println("Original combinations:")
    combinations.forEach { println("  ${it.contentToString()}") }
    
    val sorted = jline.lib.mom.util.MomUtils.sortByNnzPos(combinations)
    println("\nSorted by non-zeros:")
    sorted.forEach { println("  ${it.contentToString()}") }
    
    println("\n=== All tests completed successfully ===")
}